azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDao.java

```diff
@@ -337,7 +342,7 @@ public int selectAndUpdateExecution(final int executorId, final boolean isActive
     private static final String SELECT_EXECUTION_FOR_UPDATE_FORMAT =
         "SELECT exec_id from execution_flows WHERE status = " + Status.PREPARING.getNumVal()
             + " and executor_id is NULL and flow_data is NOT NULL and %s"
-            + " ORDER BY submit_time ASC LIMIT 1 FOR UPDATE";
+            + " ORDER BY flow_priority DESC, submit_time ASC, exec_id ASC LIMIT 1 FOR UPDATE";
```

Comment:
is there any chance that priority-based dispatching cause starvation? Should we also take waiting time into consideration?

Reply:
Thanks! It is possible right now. I'll modify the query to prevent starvation.

Reply:
Talked about this topic offline. We decided to not rollout a fix to prevent starvation of low priority executions right now. "flowPriority" feature is only available to Azkaban admins and we consider it's not very used. 
We will monitor the polling of executions to detect the moment when starvation becomes a real issue to take into consideration.
If anyone in the open source community needs a fix sooner, please refer to this commit
https://github.com/ypadron-in/azkaban/commit/0c0fe3aa2f03ad5e96b3291a20ea8937776b1f77

Reply:
thanks!